core/chains/evm/client/pool.go

```diff
@@ -195,19 +196,23 @@ func (p *Pool) ChainID() *big.Int {
 	return p.chainID
 }
 
-func (p *Pool) getRoundRobin() Node {
+func (p *Pool) getBestNode() Node {
 	nodes := p.liveNodes()
+
+	// Sorting by LatestReceivedBlockNumber, because other nodes may still be syncing
+	// TODO: https://app.shortcut.com/chainlinklabs/story/50269/add-syncing-to-liveness-checker
+	// TODO: https://app.shortcut.com/chainlinklabs/story/35230/rpc-failover-should-compare-head-to-other-nodes-in-the-pool
+	sort.Slice(nodes, func(i, j int) bool {
+		return nodes[i].LatestReceivedBlockNumber() > nodes[j].LatestReceivedBlockNumber()
```

Comment:
I think it is necessary to only call each of these just once - otherwise the value can change between comparisions.

Reply:
Agreed, I will change to find max instead of sorting, because we don't need sorting per se